resolwe/storage/models.py

```diff
@@ -243,6 +246,44 @@ def transfer_data(self, destination: "StorageLocation"):
             ReferencedPath.objects.bulk_create(referenced_paths)
             destination.files.add(*referenced_paths)
 
+    def verify_data(self) -> bool:
+        """Verify data stored in this location.
+
+        Verify hashes and sizes of all files. This operation could be slow
+        and cause network traffic, use with care.
+
+        :returns: True if data is OK, False otherwise. In case of failure
+        additional information about error is logged.
+        """
+        hash_types = ["md5", "crc32c", "awss3etag"]
+        return_value = True
+        for referenced_path in self.files:
+            url = self.get_path(filename=referenced_path.path)
+            hashes = dict(zip(hash_types, self.connector.get_hashes(url, hash_types)))
```

Comment:
Is there any use-case for `get_hashes` where you don't need such dict? If not, the method can return it.

Reply:
Not with new `transfer_objects` method, except one migration. I have changed method signature and the corresponding code.